jax_omeroutils/xml_editor.py

```diff
@@ -0,0 +1,234 @@
+import copy
+from os import sep
+from collections import defaultdict
+from ome_types.model import Project, Screen, Dataset, MapAnnotation
+from ome_types.model import DatasetRef, AnnotationRef, ImageRef
+from ome_types.model.screen import PlateRef
+from ome_types.model import CommentAnnotation, Map
+from ome_types.model.map import M
+
+CURRENT_MD_NS = 'jax.org/omeroutils/user_submitted/v0'
+
+
+def create_proj(**kwargs):
+    proj = Project(**kwargs)
+    return proj
+
+
+def create_screen(**kwargs):
+    scr = Screen(**kwargs)
+    return scr
+
+
+def create_dataset_and_ref(**kwargs):
+    ds = Dataset(**kwargs)
+    ds_ref = DatasetRef(id=ds.id)
+    return ds, ds_ref
+
+
+def create_kv_and_ref(**kwargs):
+    kv = MapAnnotation(**kwargs)
+    kvref = AnnotationRef(id=kv.id)
+    return kv, kvref
+
+
+def add_projects_datasets(ome, imp_json):
+    newome = copy.deepcopy(ome)
+    columns = imp_json['user_supplied_md']['file_metadata'][0].keys()
+    if 'project' not in columns:
+        return ome
```

Comment:
May want to also check for dataset in columns here. Likely doesn't actually matter because it won't reach this code without having both project and dataset columns given the other checks in other functions